EngineLayer/MetaMorpheusEngine.cs

```diff
@@ -72,25 +67,26 @@ public static void MatchIons(IMsDataScan<IMzSpectrum<IMzPeak>> thisScan, Toleran
             double currentTheoreticalMz = currentTheoreticalMass + Constants.protonMass;
             int testTheoreticalIndex;
             double testTheoreticalMz;
-            double testTheoreticalMass;
+
+            // speed optimizations
+            double[] experimental_mzs = thisScan.MassSpectrum.XArray;
+            double[] experimental_intensities = thisScan.MassSpectrum.YArray;
+            int numExperimentalPeaks = experimental_mzs.Length;
+
             // Loop over all experimental indices
             for (int experimentalIndex = 0; experimentalIndex < numExperimentalPeaks; experimentalIndex++)
             {
                 double currentExperimentalMz = experimental_mzs[experimentalIndex];
                 // If found match
                 if (productMassTolerance.Within(currentExperimentalMz, currentTheoreticalMz))
                 {
-                    MatchingProductsHere++;
-                    MatchingIntensityHere += experimental_intensities[experimentalIndex];
-
                     matchedIonMassesList.Add(currentTheoreticalMass);
```

Comment:
I believe this line doesn't work now, since the currentTheoreticalMass is not being updated anymore. Please either restore the code that was there in the MatchIons (with the masses), or fix this! My previous comment was about the more highly optimized CalculateClassicScore which should not have this extra mass variables.

Reply:
Great catch. I got overzealous with optimizing there.